docs/developer-guide/rules.md

```diff
@@ -25,7 +25,15 @@ You should make use of the:
 
 #### PostCSS API
 
-Use the [PostCSS API](https://api.postcss.org/) to navigate and analyze the CSS syntax tree. We recommend using the `walk` iterators (e.g. `walkDecls`), rather than using `forEach` to loop through the nodes. Use `node.raws` instead of `node.raw()` when accessing raw strings from the [PostCSS AST](https://astexplorer.net/#/gist/ef718daf3e03f1d200b03dc5a550ec60/c8cbe9c6809a85894cebf3fb66de46215c377f1a).
+Use the [PostCSS API](https://api.postcss.org/) to navigate and analyze the CSS syntax tree. We recommend using the `walk` iterators (e.g. `walkDecls`), rather than using `forEach` to loop through the nodes. If `forEach` is a better fit, you should explicitly check the `type` property of the `node` before attempting to access other properties. For example:
```

Comment:
How can I identify when `forEach` would be a better fit?

Reply:
Good question. It highlights that `forEach` is never a good fit.

I've amended the pull request to clarify:

- that checking the type is for when an array method is used
- it is quite separate from the `forEach` advice

Reply:
👍